game/state/battle/battlehazard.h

```diff
@@ -50,12 +50,13 @@ class BattleHazard : public std::enable_shared_from_this<BattleHazard>
 	            bool fireSmoke = false);
 	void grow(GameState &state);
 	void applyEffect(GameState &state);
-	void die(GameState &state, bool violently = true);
+	void die(GameState &state, bool violently, bool remove);
```

Comment:
I don't like unnamed flags changing function behaviour - it's very non-obvious when looking at the caller what "doThing(true)" and "doThing(false)" do, and easy to miss typos.

Maybe a "dieAndRemove()" call instead? Internally that can just call die() then remove?